app/models/question.rb

```diff
@@ -37,6 +37,11 @@ def create_notification; end
 
   def destroy_notification; end
 
+  def newer_submission
+    # Submissions are sorted newest first by default
+    Submission.where(id: (id + 1).., exercise_id: submission.exercise_id, course_id: course_id, user_id: submission.user_id).first
```

Comment:
```suggestion
    Submission.where(id: (submission.id + 1).., exercise_id: submission.exercise_id, course_id: course_id, user_id: submission.user_id).first
```
If I understand correctly.